### PR TITLE
Add test involving nil dynamic property values

### PR DESCRIPTION
**Pull request by padamstx**
This PR simply adds a new test which verifies that a nil dynamic property value will be correctly serialized as a JSON "null" value.

In addition to the new test, I also needed to fix an unrelated linter error found in DetailedResponse.String().

### CODE REVIEW

File: v4/core/detailed_response.go [hunk 1]
```diff
@@ -91,5 +91,5 @@ func (response *DetailedResponse) String() string {
 	if err == nil {
 		return fmt.Sprintf("%+v\n", string(output))
 	}
-	return fmt.Sprintf("Response")
```
Comment (padamstx) [inline v4/core/detailed_response.go:94]: I saw a linter error on this statement due to the unnecessary use of fmt.Sprintf().   This is the first go core PR I've worked on after upgrading my laptop to Fedora 32 (the latest) which uses go version 1.14.3.   Prior to this, I was running Fedora 30, which uses go 1.12.x.   I'm guessing that the upgrade to the latest Go engine is why I'm now seeing this linter error, when it wasn't seen before.   Anyway, I fixed it :)
